activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -263,6 +263,11 @@ def to_hash
       _new_hash
     end
 
+    def ==(other)
+      super(other.try(:with_indifferent_access))
```

Comment:
This will create two hashes with indifferent access just to do the comparing. One here and other in the `super`. We should implement the logic here:

```ruby
def ==(other)
  if other.is_a?(HashWithIndifferentAccess)
    to_hash == other.to_hash
  else
    to_hash == other
  end
end
```